app/ssttpcalculator/legacy/CalculatorService.scala

```diff
@@ -96,6 +96,13 @@ class CalculatorService @Inject() (
   def closestSchedule(amount: BigDecimal, schedules: Seq[PaymentSchedule]): Option[PaymentSchedule] = {
       def difference(schedule: PaymentSchedule) = math.abs(schedule.getMonthlyInstalment.toInt - amount.toInt)
       def closest(min: PaymentSchedule, next: PaymentSchedule) = if (difference(next) < difference(min)) next else min
+
+    schedules.reduceOption(closest)
+  }
+
+  def closestScheduleEqualOrLessThan(amount: BigDecimal, schedules: Seq[PaymentSchedule]): Option[PaymentSchedule] = {
```

Comment:
you could probably avoid some duplication by implementing this in terms of the method above like:
```scala
def closestScheduleEqualOrLessThan(amount: BigDecimal, schedules: Seq[PaymentSchedule]): Option[PaymentSchedule] =
  closestSchedule(
    amount,
    schedules.filter(schedule => amount - schedule.instalmentAmount >= 0)
  )
```